packages/cloudflare/src/entrypoints/image-service.ts

```diff
@@ -0,0 +1,40 @@
+import type { ExternalImageService } from 'astro';
+
+import { baseService } from 'astro/assets';
+import { isESMImportedImage, isRemoteAllowed, joinPaths } from '../utils/assets.js';
+
+const service: ExternalImageService = {
+	validateOptions: baseService.validateOptions,
+	getHTMLAttributes: baseService.getHTMLAttributes,
+	getSrcSet: baseService.getSrcSet,
+	getURL: (options, imageConfig) => {
+		const resizingParams = [];
+		if (options.width) resizingParams.push(`width=${options.width}`);
+		if (options.height) resizingParams.push(`height=${options.height}`);
+		if (options.quality) resizingParams.push(`quality=${options.quality}`);
+		if (options.fit) resizingParams.push(`fit=${options.fit}`);
+		if (options.format) resizingParams.push(`format=${options.format}`);
+
+		let imageSource = '';
+		if (isESMImportedImage(options.src)) {
+			imageSource = options.src.src;
+		} else if (isRemoteAllowed(options.src, imageConfig)) {
+			imageSource = options.src;
+		} else {
+			// If it's not an imported image, nor is it allowed using the current domains or remote patterns, we'll just return the original URL
+			return options.src;
+		}
```

Comment:
This is more of a thought on `astro:assets`, but I hate that you have to reproduce this logic here. It's flexible, but it's very inconvenient. I think I'll expose something eventually to check if an URL is authorized.

Reply:
Oh yeah that would be nice, if some of those util/helper functions are exposed :)